R/render.r

```diff
@@ -422,7 +422,121 @@ data_deps <- function(pkg, depth) {
       )
     )
   )
-  deps <- bslib::bs_theme_dependencies(bs_theme)
+
+  # map either secondary OR fg, bg to component-active-bg
+  # and also dropdown-link-active-bg
+  # unless a value was set by the user
+  if (!is.null(pkg$meta[["template"]]$bslib$secondary)) {
+    if (is.null(pkg$meta[["template"]]$bslib$`component-active-bg`)) {
+      bs_theme <- bslib::bs_add_variables(
+        bs_theme,
+        "component-active-bg" = as.character(
+          bslib::bs_get_variables(bs_theme, "secondary")
+        )
+      )
+    }
+  } else {
+    if (is.null(pkg$meta[["template"]]$bslib$`component-active-bg`)) {
+      bs_theme <- bslib::bs_add_variables(
+        bs_theme,
+        "component-active-bg" = "mix($body-color, $body-bg, 5%)",
+        .where = "declarations"
+      )
+    }
+    if (is.null(pkg$meta[["template"]]$bslib$`dropdown-link-active-bg`)) {
+      bs_theme <- bslib::bs_add_variables(
+        bs_theme,
+        "dropdown-link-active-bg" = "mix($body-color, $body-bg, 5%)",
+        .where = "declarations"
+      )
+    }
+  }
+
+  # map body color to navbar colors
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-color" = "rgba($body-color, 0.8)",
+      .where = "declarations"
+    )
+  }
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-hover-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-hover-color" = "rgba($body-color, 0.9)",
+      .where = "declarations"
+    )
+  }
+
+  # map primary if set, or fg otherwise
+  # to navbar-light-active-color + component-active-color
+  # unless a value was set by the user
+  default_prim_color <- if (!is.null(pkg$meta[["template"]][["bslib"]][["primary"]])) {
+    pkg$meta[["template"]][["bslib"]][["primary"]]
+  } else {
+    as.character(bslib::bs_get_variables(bs_theme, "fg"))
+  }
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-active-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-active-color" = default_prim_color
+    )
+  }
```

Comment:
I'm pretty sure you can avoid all the `if (is.null())` checking as long as you define the pkgdown defaults with a `!default` flag, then allow the users settings to override those defaults:

```r
pkgdown_defaults <- list(foo = "bar !default")
theme <- bs_add_variables(theme, !!!pkgdown_defaults)
theme <- bs_add_variables(theme, template_vars(pkg$meta$template))
```